### DISCUSSION TITLE
TextLayer outline does not work

### DISCUSSION THREAD

**Discussion by pequeurb** [visgl/deck.gl#6022] [Q&A] [answered]
I created a new class that extends from CompositeLayer, in which I use some IconLayers and one TextLayer.
Everything is working fine, but the text is not always very readable on the map, so I wanted to add a white outline around the text with some shadows to fix this.

Something like this:
![image](https://user-images.githubusercontent.com/87120382/127156724-418e135a-aed9-418d-b690-a2b717db9e1f.png)

I saw that there are options available for the text outline, but nothing for shadows.
However, if I try to use outlineWidth and outlineColor, nothing happens.
If I'm reading the documentation on TextLayer correctly, I need to enable the fontsetting "sdf".

This is the TextLayer I'm using:
```
new TextLayer(
  this.getSubLayerProps({
    id: "text-layer",
    data: this.props.data,

    fontFamily: this.props.fontFamily,
    fontWeight: this.props.fontWeight,
    maxWidth: this.props.maxWidth,
    wordBreak: this.props.wordBreak,

    fontSettings: {
      sdf: true, // needed for text outline
    },
    outlineWidth: 20,
    outlineColor: [255, 0, 255, 255],

    getPosition: this.props.getPosition,
    getText: this.props.getText,
    getSize: this.props.getTextSize,
    getColor: this.props.getTextColor,
    getAngle: this.props.getTextAngle,
    getTextAnchor: this.props.getTextAnchor,
    getAlignmentBaseline: this.props.getTextAlignmentBaseline,
    getPixelOffset: this.props.getTextPixelOffset,

    updateTriggers: {
      getPosition: this.props.updateTriggers.getPosition,
      getText: this.props.updateTriggers.getText,
      getSize: this.props.updateTriggers.getTextSize,
      getColor: this.props.updateTriggers.getTextColor,
    },
  })
),
```

The only thing that "fontSettings.sdf: true" accomplishes is that it makes the text even more unreadable by putting some weird "worn off" effect on it. And I don't get any outline, no matter what color or width I try.
![image](https://user-images.githubusercontent.com/87120382/127158068-49016fb6-72b0-4d1a-812f-f0b77f4e3b31.png)

This is what I had before adding fontSettings or OutlineWidth/Color:
![image](https://user-images.githubusercontent.com/87120382/127159418-f284b6a3-49ad-4a23-ac29-481a9d056d68.png)

I tried playing around with the other fontSetting options like fontSize, smoothing or cutoff, but it doesn't change my situation much.
Am I doing something wrong or is this a bug?

**Comment by Pessimistress** [comment, marked as answer]
The outline feature is available in v8.5.0 and above.

For sdf font readability, you should try adjusting the other options in `fontSettings`.

**Reply by pequeurb** [reply]
I didn't realize it was such a recent feature, it works now, thanks!
As for the readability, it also improved after updating (was using 8.4.8).
One thing I noticed is that all icons and text have almost doubled in size after the update without changing any code. 
Not sure why, but I guess I can update the sizes where necessary, I just figured I'd mention it. 😉

**Reply by Pessimistress** [reply]
You should visit the [upgrde guide](https://deck.gl/docs/upgrade-guide#upgrading-from-deckgl-v84-to-v85)

**Reply by pequeurb** [reply]
That explains a few things, thanks again!
